staging/src/k8s.io/client-go/util/workqueue/parallelizer.go

```diff
@@ -61,3 +63,43 @@ func ParallelizeUntil(ctx context.Context, workers, pieces int, doWorkPiece DoWo
 	}
 	wg.Wait()
 }
+
+// StaticParallelizeUntil is a framework that allows for parallelizing N
+// independent pieces of work until done or the context is canceled.
+// Unlike ParallelizeUntil, it assigns the load to each routine statically.
+func StaticParallelizeUntil(ctx context.Context, workers, pieces int, doWorkPiece DoWorkPieceFunc) {
```

Comment:
Rather than do this, you can make the other function send "chunks" of work (e.g., N pieces per channel write/read). That's the best strategy I personally have found for keeping a group of goroutines all busy to get the smallest wall time.

I've never had a problem where things were "even" enough that this strategy didn't leave one goroutine running a fair amount after others had finished.

Reply:
That's an alternative I was thinking of too. However, it might be difficult to come up with a good chunk size. It can be paramaterized but then we have to go through each call and find the appropriate chunk size.

On the other hand, perhaps a reasonable chunk size is sqrt(N).

Reply:
I would just parameterize it.

Existing call sites can just pass `1` as the chunk size, that's what they're doing today. I would leave the existing function signature unchanged, and make it call a function which has this parameter for the implementation.

Also add a test :)

Reply:
I'll add `ParallelizeUntilWithChunks` that uses `ParallelizeUntil` underneath.

> Also add a test :)

For sure, I was gathering feedback first :D

Reply:
adding options to the currently function is also an option (e.g., WithChunkSize)

Reply:
It looks like `BenchmarkSchedulingPodAntiAffinity` has a very low load. This is testing **required** pod antiaffinity, so we usually have <= 1 pods per node. Even when using chunks, the channel management is not negligible.
In summary:
- channels (original): 16ms
- channels and chunk size of sqrt(N) or N/W: 15ms
- Static partitioning: 6ms

Now, for `BenchmarkSchedulingPreferredPodAffinity`, the benefits are more noticeable. Since this tests **preferred**, we have more pods per node.
- channels (original): 25ms
- channels and chunk size of sqrt(N) or N/W: 15ms
- Static partitioning: 7ms

You could argue that our benchmarks are too synthetic (all nodes look very similar). We could improve this. But I think there are still arguments for doing static partitioning. Perhaps, we can try to randomize the assignment a bit.

Reply:
That's super surprising. Static partitioning should be equivalent to a particular chunk size, no? What is N and W in your case?

Reply:
Can you publish the code you used to produce those numbers? The static routine in this commit has a bug that causes it to (probably) do the last chunk N times.

Reply:
It looks like it was causing it to do the work that the last routine was supposed to get, which was the remainder of # nodes / # threads.

After fixing it, the numbers of static partitioning and chunk match. I'll update the PR soon.

Reply:
Awesome :)

It's a running joke in my team that you can get a large performance increase with a well-placed `return` statement, this is similar ;)